packages/reactor/docs/planning/Operations/upgrades.md

```diff
@@ -0,0 +1,76 @@
+# Upgrades
+
+### Summary
+
+Document Model upgrades provide a systematic approach to versioning and backward compatibility.
+
+In Reactor, Actions act much like typical Command objects in the Event Sourcing architecture and Operations act much like Events.
+
+However, unlike typical ES approaches, Operations _are never upcasted_. That is, version upgrades never mutate Operations. Instead, every `Action` must be executed by a specific version of the corresponding document model library to result in the expected `Operation`.
+
+Operations are then decoupled from versioning, as the `Action` holds this information.
+
+### Document Creation Flow
+
+Suppose we have a Document Model of type `ph/todo`.
+
+1. First, we create a new document:
+
+```tsx
+import { createDocument } from "ph/todo";
+
+const doc = await createDocument({
+  title: "My Todo List",
+});
+```
+
+2. Next, we submit it to the `IReactor`.
+
+```tsx
+
+const jobStatus = await reactor.create(doc);
+
+```
+
+3. Internally, the `IReactor` will:
+
+- Create an `Action` of type `ACTION_CREATE`.
+- Create an `Action` of type `ACTION_UPGRADE`.
```

Comment:
If we want to follow the imperative naming we have been using for operations, this could be `CREATE_DOCUMENT` and `UPGRADE_DOCUMENT` instead.